src/GafferUI/StandardNodeGadget.cpp

```diff
@@ -645,6 +645,13 @@ Gadget *StandardNodeGadget::closestDragDestinationProxy( const DragDropEvent &ev
 		return nullptr;
 	}
 
+	// gadgets on the same node are invalid destinations and don't support snapping
+	ConnectionCreator *creator = IECore::runTimeCast<ConnectionCreator>( event.sourceGadget.get() );
+	if( creator && isAncestorOf( creator ) )
```

Comment:
Should this be `if( !creator || isAncestorOf( creator ) )`? Below we are definitely assuming that `creator` is non-null...

Reply:
Testing github. Please Ignore.